spark/spark-tensorflow-distributor/spark_tensorflow_distributor/mirrored_strategy_runner.py

```diff
@@ -0,0 +1,332 @@
+import logging
+import math
+import os
+import random
+import re
+import sys
+
+
+from pyspark.sql import SparkSession
+
+
+def _get_logger(name):
+    """
+    Gets a logger by name, or creates and configures it for the first time.
+    """
+    logger = logging.getLogger(name)
+    logger.setLevel(logging.INFO)
+    # If the logger is configured, skip the configure
+    if not logger.handlers and not logging.getLogger().handlers:
+        handler = logging.StreamHandler(sys.stderr)
+        logger.addHandler(handler)
+    return logger
+
+
+class MirroredStrategyRunner:
+    """
+    MirroredStrategyRunner runs TensorFlow deep learning training jobs on Spark clusters.
+    It trains synchronously by mirroring the model's variables among all the workers and
+    shares the gradient updates in a decentralized manner. MirroredStrategyRunner can take
+    a regular TensorFlow program with no special distributed training code and launch it
+    as a distributed training program.
+
+    .. note:: See more at https://www.tensorflow.org/guide/distributed_training
+    """
+    def __init__(self, num_slots, gpu_resource_name='gpu', use_custom_strategy=False):
```

Comment:
Since we have more params now, shall we use `def __init__(self, *, num_slots, ...)` to enforce keyword args?